shipment_planning_maxoptra/wizard/shipment_maxoptra_schedule_import.py

```diff
@@ -138,33 +149,51 @@ def regroup_operations(
     def create_delivery_batch_picking_by_vehicle(self, schedule_by_vehicles):
         batch_ids = []
         for vehicle_name, maxoptra_deliveries in schedule_by_vehicles.items():
-            batch_picking_values = self._prepare_batch_picking_values(
-                vehicle_name, driver_name=maxoptra_deliveries[0].get("driver")
-            )
-            batch_picking = self.env["stock.picking.batch"].create(batch_picking_values)
-            batch_ids.append(batch_picking.id)
-            for maxoptra_delivery in maxoptra_deliveries:
-                picking = self.env["stock.picking"].search(
-                    [("name", "=", maxoptra_delivery.get("picking_name"))]
+            for p_type, picks in self._prepare_pickings(maxoptra_deliveries):
+                batch_picking_values = self._prepare_batch_picking_values(
+                    vehicle_name, driver_name=maxoptra_deliveries[0].get("driver")
                 )
-                # TODO: Add more checks?
-                if not picking:
-                    raise UserError(
-                        _("No matching picking found for Order reference %s")
-                        % maxoptra_delivery.get("picking_name")
+                batch_picking = self.env["stock.picking.batch"].create(batch_picking_values)
+                batch_ids.append(batch_picking.id)
+                for pick in picks:
+                    pick.write(
+                        {
+                            "batch_id": batch_picking.id,
+                            "vehicle_id": batch_picking.vehicle_id.id,
+                            "driver_id": batch_picking.driver_id.id,
+                        }
                     )
```

Comment:
IMO we don't need this loop anymore if scheduled time per picking was already updated beforehand and we write the same values on each picking.